src/components/ListView.jsx

```diff
@@ -50,21 +48,54 @@ function ListView() {
   //
   const handleCheckboxChange = async (e) => {
     const itemId = e.target.name;
-    //create time variable to save time when user checked the box as purchasedTime
-    const datePurchased = Date.now();
-    //if user want to uncheck the item it can be done and purchasedDate is set to null again
+    // grabbing the specific item from state that is clicked because we will be updating its properties
+    const itemToUpdate = items.find((item) => itemId === item.itemName);
+
+    let daysSinceLastTransaction;
+
+    // if user checks a box, itemToUpdate is taken through this flow
     if (e.target.checked) {
-      setUpdateToDb(localToken, itemId, 'purchasedDate', datePurchased);
+      // check to see if the item has never been purchased before (purchasedDate is set to null by default)
+      if (!itemToUpdate.purchasedDate) {
+        daysSinceLastTransaction =
```

Comment:
actually, do we need this if statement? I think it is no need for it

Reply:
So I think we need both daysSinceLastTransaction because purchasedDate is set to null initially. I'm not sure about saving purchasedDate when an item is created because that is kind of misleading - we could call it something else, but an item added to the list has not been purchased yet.

Reply:
Ah, that makes sense. I guess my only concern is how are naming things. I'd be curious to know what other folks are thinking about this as well!

Reply:
I am confused about the purchasedDate part being the date they add an item to their list @alenamedved 

My assumption that a user can add an item and it doesn't have to be the day they purchased it on because one of the very first issues had us set the initial purchaseDate to null. For example if a user was compiling their list to be ready to start using it before they make any purchases because they just want to get organized at 9pm on a Saturday :P It might just be me but I personally wouldn't add to my inventory list as I buy things, I would do it in advance to stay organized.

 I'm imagining that this purchase date and prediction is leading into notifying users in a later feature so if it's not null and has whatever date they added the item, even if that date isn't the date they purchased it then they may get a false notification and from reading the code it seems like there's a purchased date check which controls the algorithm via this if statement which would just make our data more accurate and ensure that we are controlling the date based off a specific user action - the checkboxes.

Reply:
@lindseyindev 
yes, looks like the app will be notifying users using previousEstimate as a break point
But I don't see how if can cause a false notification. When item added for the first time it added with previousEstimate number = frequency which corresponds to radio button clicked by user. 
When item checked for the first time - previousEstimate = daysSinceLastTransaction (currentDay - createdAt) //it is where I believe we can save createdAt as purchasedDate, because it won't affect calculations, but makes the code cleaner and easier to read, and we don't have to save extra field in fb. Maybe it is not really matter as long as the code works, I'm just interested to find out if it can be done better..)
When second time checked - previousEstimate = calculateEstimate()
And for those how reads the code we can leave a comment that if totalPurchases = 0 that means that purchasedDate is the date when item was added.